test/views/pages/home/dynamic_home_test.dart

```diff
@@ -0,0 +1,100 @@
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:mockito/mockito.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/services/sorting/post_sort_option.dart";
+import "package:proxima/services/sorting/post_sorting_service.dart";
+import "package:proxima/viewmodels/feed_sort_options_view_model.dart";
+import "package:proxima/viewmodels/home_view_model.dart";
+import "package:proxima/views/home_content/feed/feed_sort_option_chips.dart";
+import "package:proxima/views/pages/home/home_page.dart";
+
+import "../../../mocks/data/firestore_post.dart";
+import "../../../mocks/data/firestore_user.dart";
+import "../../../mocks/data/geopoint.dart";
+import "../../../mocks/providers/provider_homepage.dart";
+import "../../../mocks/services/mock_geo_location_service.dart";
+
+void main() {
+  const position = userPosition0;
+
+  late FakeFirebaseFirestore firestore;
+  late MockGeoLocationService geoLocationService;
+  late ProviderScope homepageWidget;
+  late List<PostFirestore> posts;
+
+  setUp(() async {
+    firestore = FakeFirebaseFirestore();
+    geoLocationService = MockGeoLocationService();
+    homepageWidget = homePageFakeFirestoreProvider(
+      firestore,
+      geoLocationService,
+    );
+
+    when(geoLocationService.getCurrentPosition()).thenAnswer(
+      (_) => Future.value(position),
+    );
+
+    final locations = GeoPointGenerator.generatePositions(position, 20, 0);
+
+    final postGenerator = FirestorePostGenerator();
+    posts = postGenerator.generatePostsAtDifferentLocations(
+      locations,
+    );
+    await setPostsFirestore(posts, firestore);
+
+    final users = FirestoreUserGenerator.generateUserFirestoreWithId(
+      posts.map((post) => post.data.ownerId).toList(),
+    );
+    await setUsersFirestore(users, firestore);
+  });
+
+  group("Sort options", () {
+    testWidgets(
+      "Pushing the sort option button sets the correct order to the posts",
+      (tester) async {
+        await tester.pumpWidget(homepageWidget);
+        await tester.pumpAndSettle();
+
+        final element = tester.element(find.byType(HomePage));
+        final container = ProviderScope.containerOf(element);
```

Comment:
I feel that we can find a better way to get the providerScope and read results from the `feedSortOptionsProvider`. 

We can maybe take inspiration from @yoannLafore's repository testing, where it creates a `ProviderContainer` and reads the provider from it. ([ref](https://github.com/ProximaEPFL/proxima/pull/181/files#diff-9f98d592c3821b134d5ac116b3644a38358e76cd72a4b974d6468c572e7a04e3)).

I don't know if this the best way, but I found a way to make this idea working:
I create the homeWidget with: 
```dart 
homepageWidget = UncontrolledProviderScope(
  container: containerProvider,
  child: homePageApp,
);

 ```
where 
```dart
containerProvider = ProviderContainer(overrides: [
  firestoreProvider.overrideWithValue(firestore),
  geoLocationServiceProvider.overrideWithValue(geoLocationService),
]);

 ```
 
Using that, I am able to use: 
```dart 
expect(
  containerProvider.read(feedSortOptionsProvider),
  equals(sortOption),
);

// Expect the list to be in correct order
// We compare the post id values for an easier debugging.
final actualPostsOverview = await containerProvider.read(
  postOverviewProvider.future,
);


```

But since we never used `UncontrolledProviderScope` in our codebase, I am not sure if this is a better solution.

Reply:
By humble curiosity, why do you call this a "better way"? This looks like more boilerplate to me at first sight.

Reply:
I feel like we are extracting the container from the UI, when we could have extracted it more globally without going through the UI, but I may be wrong.

Reply:
What I did is what is recommended [in the documentation](https://riverpod.dev/docs/essentials/testing) for testing both widgets and providers at the same time. In my opinion, this is ok.

Reply:
You are right. Thank you.